activesupport/lib/active_support/core_ext/class/subclasses.rb

```diff
@@ -28,10 +28,9 @@ def descendants
   #
   #   Foo.subclasses # => [Bar]
   def subclasses
-    subclasses, chain = [], descendants
-    chain.each do |k|
-      subclasses << k unless chain.any? { |c| c > k }
+    chain = descendants
+    chain.reject do |k|
+      chain.any? { |c| c > k }
     end
```

Comment:
I wonder if there's a way to avoid the double loops by keeping a hash lookup here or using things like `&` or `|`. Though nothing springs to mind.

Reply:
I was also hoping to do something like that, but couldn't come up with anything either. It's my hope that my refactoring at least makes this double looping behaviour more obvious.

Reply:
I found the solution in a new commit!